Prefer the plugin with the correct suffix for the current compile mode.
---
diff --git a/src/qml/qml/qqmlpluginimporter.cpp b/src/qml/qml/qqmlpluginimporter.cpp
index 24f12891..86ec1c48 100644
--- a/src/qml/qml/qqmlpluginimporter.cpp
+++ b/src/qml/qml/qqmlpluginimporter.cpp
@@ -360,11 +360,11 @@ QString QQmlPluginImporter::resolvePlugin(const QString &qmldirPluginPath, const
     static const QString prefix;
     static const QStringList suffixes = {
     # ifdef QT_DEBUG
-        QLatin1String("d.dll"), // try a qmake-style debug build first
+        QLatin1String("_d.dll"), // try a qmake-style debug build first
         QLatin1String(".dll")
-    #else
+    # else
         QLatin1String(".dll"),
-        QLatin1String("d.dll") // try a qmake-style debug build after
+        QLatin1String("_d.dll") // try a qmake-style debug build after
     # endif
     };
 #elif defined(Q_OS_DARWIN)
@@ -386,7 +386,13 @@ QString QQmlPluginImporter::resolvePlugin(const QString &qmldirPluginPath, const
     # if defined(Q_OS_ANDROID)
         QStringLiteral(LIBS_SUFFIX),
     # endif
+    # ifdef QT_DEBUG
+        QLatin1String("_d.so"), // try a qmake-style debug build first
         QLatin1String(".so")
+    # else
+        QLatin1String(".so"),
+        QLatin1String("_d.so") // try a qmake-style debug build after
+    # endif
     };
 #endif
 
